hipcheck/src/metric/entropy.rs

```diff
@@ -150,8 +152,70 @@ fn is_likely_source_file(
 
 /// Calculate grapheme frequencies for each commit.
 fn grapheme_freqs(commit_diff: &CommitDiff, db: &dyn MetricProvider) -> Result<CommitGraphemeFreq> {
-	let grapheme_freqs = GraphemeFreqCalculator::for_diff(&commit_diff.diff, db)?.calculate();
+	// Dashmap (fast concurrent hashmap) to store counts for each grapheme.
+	let grapheme_table: DashMap<String, u64> = DashMap::new();
 
+	// Iterate over the file diffs by reference.
+	let mut res: Vec<usize> = vec![];
+	commit_diff
+		.diff
+		.file_diffs
+		.iter()
+		.filter(|file_diff| {
+			// Filter out any that are probably not source files, or are empty patches
+			match db.is_likely_source_file(Arc::clone(&file_diff.file_name)) {
+				Ok(true) => file_diff.patch.is_empty().not(),
+				_ => false,
+			}
+		})
```

Comment:
This section is the reason I avoided parallelizing over files -- any `Err` returned here just gets filtered out of the calculation rather than being reported.

Reply:
Just pushed a fix to this (swapping `filter` for `filter_map`) to propagate errors from `db.is_likely_source_file`. and return an error before we get to grapheme calculation